src/genie/libs/parser/iosxe/cat9k/c9400/show_platform.py

```diff
@@ -178,30 +178,34 @@ class ShowEnvironmentAllSchema(MetaParser):
                 }
             }
         },
-        'power_supply': {
-            'slot': {
-                Any(): {
-                    'model_no': str,
-                    'type': str,
-                    'capacity': str,
+        'switch': {
+            Any(): {
+                'power_supply': {
+                    'slot': {
+                        Any(): {
+                            'model_no': str,
+                            'type': str,
+                            'capacity': str,
+                            'status': str,
+                            'fan_1_state': str,
+                            'fan_2_state': str,
+                        }
+                    },
+                    'current_configuration_mode': str,
+                    'current_operating_state': str,
+                    'currently_active': int,
+                    'currently_available': int,
+                },
```

Comment:
A bit of an issue. Due to our strict adherence to backwards compatibility we can't change schemas like this. It has the potential to break older testscripts which we cannot have happen.

We do have a concept of revisions to address this however. It allows us to introduce new revisions of parsers without breaking backwards compatibility. 

You can read up on them here https://pubhub.devnetcloud.com/media/pyats-development-guide/docs/writeparser/writeparser.html#revising-a-parser